community/bolt/src/main/java/org/neo4j/bolt/v1/messaging/BoltRequestMessageReader.java

```diff
@@ -89,20 +80,20 @@ public void read( BoltRequestMessageHandler handler ) throws IOException
                     handler.onPullAll();
                     break;
                 default:
-                    throw new BoltIOException( Status.Request.Invalid,
-                            "Message 0x" + Integer.toHexString( signature ) + " is not supported." );
+                    throw new BoltIOException( Status.Request.InvalidFormat,
+                            String.format( "Message 0x%s is not supported.", Integer.toHexString( signature ) ) );
```

Comment:
These error messages are not very helpful for us when a user come to us with such an error in their stack trace. We usually have to ask what the user have done. So maybe it would be useful and possible after this PR that we could print out the whole message that failed to decode. So that we could get a clue what did the user send.

This is a good-to-have but not need-to-fix in this PR thing.

Reply:
I'm carding this.